pages/history.tsx

```diff
@@ -0,0 +1,222 @@
+      setInvalid(true);
+      return;
+    }
+
+    (async () => {
+      if (!state.currentContract) return;
+
+      let c = await state.connection.contract.at(state.currentContract, tzip16);
+      let balance = await state.connection.tz.getBalance(state.currentContract);
+
+      let cc = await c.storage();
+      let version = await (state.contracts[state.currentContract]
+        ? Promise.resolve<version>(
+            state.contracts[state.currentContract].version
+          )
+        : fetchVersion(c));
+      const updatedContract = toStorage(version, cc, balance);
+      state.contracts[state.currentContract]
+        ? dispatch({
+            type: "updateContract",
+            payload: {
+              address: state.currentContract,
+              contract: updatedContract,
+            },
+          })
+        : null;
+      let bigmap: { key: string; value: any }[] = await getProposals(
+        getProposalsId(version, cc)
+      );
+      let transfers = await getTransfers(state.currentContract);
+      let proposals: [number, any][] = bigmap.map(({ key, value }) => [
+        Number.parseInt(key),
+        { ui: toProposal(version, value), og: value },
+      ]);
+      setContract(updatedContract);
+      setTransfers(transfers);
+      setProposals(proposals);
+      setIsLoading(false);
+    })();
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [state.currentContract]);
+
+  const filteredProposals = useMemo(
+    () => [
+      ...proposals.filter(
+        ([_, proposal]) => !("Proposing" === proposal.ui.status)
+      ),
+    ],
+    [proposals]
+  );
+
+  return (
+    <div className="min-h-content relative flex grow flex-col">
+      <Meta title={"Wallets"} />
+      <Modal opened={!!openModal.state}>
+        {!!openModal.state && (
+          <ProposalSignForm
+            address={state.currentContract ?? ""}
+            threshold={contract.threshold}
+            version={contract.version}
+            proposal={proposals.find(x => x[0] === openModal.proposal[1])![1]}
+            state={openModal.proposal[0]}
+            id={openModal.proposal[1]}
+            closeModal={() => setCloseModal((s: any) => ({ ...s, state: 0 }))}
+          />
+        )}
+      </Modal>
+      <div>
+        <div className="mx-auto flex max-w-7xl justify-start py-6 px-4 sm:px-6 lg:px-8">
+          <h1 className="text-2xl font-extrabold text-white">History</h1>
+        </div>
+      </div>
+      <main className="h-full min-h-fit grow">
+        <div className="mx-auto h-full min-h-full max-w-7xl py-6 px-4 sm:px-6 lg:px-8">
+          {invalid ? (
+            <div className="mx-auto flex w-full items-center justify-center bg-graybg p-2 shadow">
+              <p className="mx-auto text-xl font-bold text-gray-800">
+                Invalid contract address: {state.currentContract}
+              </p>
+            </div>
+          ) : isLoading ? (
```

Comment:
Currently, it looks like if the `state.currentContract` is null, the `invalid` is null so the page will keep loading. so as the other pages.

Reply:
![image](https://user-images.githubusercontent.com/115862/221080088-abdffe99-8d04-47fd-826d-352d00707ca2.png)
It looks a bit weird but let's keep it for now. I'm thinking the other idea, let's discuss offline.

Reply:
![image](https://user-images.githubusercontent.com/115862/221080088-abdffe99-8d04-47fd-826d-352d00707ca2.png)
Can you change it "please import/create a wallet" for now?

(I'm thinking the other idea, let's discuss offline.)

Reply:
I added the redirection when I don't detect anymore wallets or user is unauthanticated